lib/modules/platform/codecommit/codecommit-client.ts

```diff
@@ -0,0 +1,344 @@
+import {
+  CodeCommitClient,
+  CreatePullRequestApprovalRuleCommand,
+  CreatePullRequestApprovalRuleInput,
+  CreatePullRequestApprovalRuleOutput,
+  CreatePullRequestCommand,
+  CreatePullRequestInput,
+  CreatePullRequestOutput,
+  DeleteCommentContentCommand,
+  DeleteCommentContentInput,
+  DeleteCommentContentOutput,
+  DescribePullRequestEventsCommand,
+  DescribePullRequestEventsInput,
+  DescribePullRequestEventsOutput,
+  GetCommentsForPullRequestCommand,
+  GetCommentsForPullRequestInput,
+  GetCommentsForPullRequestOutput,
+  GetFileCommand,
+  GetFileInput,
+  GetFileOutput,
+  GetPullRequestCommand,
+  GetPullRequestInput,
+  GetPullRequestOutput,
+  GetRepositoryCommand,
+  GetRepositoryInput,
+  GetRepositoryOutput,
+  ListPullRequestsCommand,
+  ListPullRequestsInput,
+  ListPullRequestsOutput,
+  ListRepositoriesCommand,
+  ListRepositoriesInput,
+  ListRepositoriesOutput,
+  MergeBranchesByFastForwardCommand,
+  MergeBranchesByFastForwardInput,
+  MergeBranchesByFastForwardOutput,
+  MergeBranchesBySquashCommand,
+  MergeBranchesBySquashInput,
+  MergeBranchesBySquashOutput,
+  PostCommentForPullRequestCommand,
+  PostCommentForPullRequestInput,
+  PostCommentForPullRequestOutput,
+  PullRequestEventType,
+  PullRequestStatusEnum,
+  UpdateCommentCommand,
+  UpdateCommentInput,
+  UpdateCommentOutput,
+  UpdatePullRequestDescriptionCommand,
+  UpdatePullRequestDescriptionInput,
+  UpdatePullRequestDescriptionOutput,
+  UpdatePullRequestStatusCommand,
+  UpdatePullRequestStatusInput,
+  UpdatePullRequestStatusOutput,
+  UpdatePullRequestTitleCommand,
+  UpdatePullRequestTitleInput,
+  UpdatePullRequestTitleOutput,
+} from '@aws-sdk/client-codecommit';
+import type { Credentials } from '@aws-sdk/types';
+import is from '@sindresorhus/is';
+import * as aws4 from 'aws4';
+import { REPOSITORY_UNINITIATED } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+
+let codeCommitClient: CodeCommitClient;
+
+export function buildCodeCommitClient(
+  region: string,
+  credentials: Credentials
+): CodeCommitClient {
+  if (!codeCommitClient) {
+    codeCommitClient = new CodeCommitClient({
+      region,
+      credentials,
+    });
+  }
+  return codeCommitClient;
+}
+
+export async function deleteComment(
+  commentId: string
+): Promise<DeleteCommentContentOutput> {
+  const input: DeleteCommentContentInput = {
+    commentId,
+  };
+  const cmd = new DeleteCommentContentCommand(input);
+  return await codeCommitClient.send(cmd);
```

Comment:
it seems all functions will throw null reference exceptions when the client wasn't build. seems bad code design to me.

Reply:
it throws an exception and stops the run if it wasn't initialized, ill make another precaution, in case it passed with undefined

Reply:
sure, maybe this should be refactored to a class where the client is a private property initialized in constructor. the class instance can be create inside init platform.

Reply:
This requires a major refactor that will delay release. Since this implementation is a first step in making a more complete integration in the future, I think we should release it like that and keep in mind to refactor when the complete integration will be developed.
Currently it works fine and I added extra checks and tests.

@betterPeleg fyi